internal/pkg/acl/acl.go

```diff
@@ -317,86 +315,52 @@ func CreateAclRequestDataToAclData(data *AclRequestDataWithError) cpkafkarestv3.
 	}
 }
 
-func PrintACLsFromKafkaRestResponseWithResourceIdMap(cmd *cobra.Command, acls []cckafkarestv3.AclData, numericIdToResourceId map[int32]string) error {
-	list := output.NewList(cmd)
-	for _, acl := range acls {
-		prefix, resourceId, err := getPrefixAndResourceIdFromPrincipal(acl.Principal, numericIdToResourceId)
-		if err != nil {
-			if err.Error() == errors.UserIdNotValidErrorMsg {
-				continue // skip the entry if not a valid user id
-			}
-			return err
-		}
-		list.Add(&out{
-			Principal:    prefix + ":" + resourceId,
-			Permission:   acl.Permission,
-			Operation:    acl.Operation,
-			ResourceType: string(acl.ResourceType),
-			ResourceName: acl.ResourceName,
-			PatternType:  acl.PatternType,
-		})
+func PrintACLsFromKafkaRestResponse(cmd *cobra.Command, acls []cckafkarestv3.AclData) error {
+	showAll, err := cmd.Flags().GetBool("all")
+	if err != nil {
+		return err
 	}
-	list.Filter(listFields)
-	return list.Print()
-}
 
-func PrintACLsWithResourceIdMap(cmd *cobra.Command, acls []*ccstructs.ACLBinding, numericIdToResourceId map[int32]string) error {
 	list := output.NewList(cmd)
 	for _, acl := range acls {
-		prefix, resourceId, err := getPrefixAndResourceIdFromPrincipal(acl.Entry.Principal, numericIdToResourceId)
-		if err != nil {
-			if err.Error() == errors.UserIdNotValidErrorMsg {
-				continue // skip the entry if not a valid user id
+		principal := acl.GetPrincipal()
+		if !showAll {
+			if integerId, err := principalHasIntegerID(principal); err != nil {
+				return err
+			} else if integerId {
+				continue
 			}
-			return err
 		}
 		list.Add(&out{
-			Principal:    prefix + ":" + resourceId,
-			Permission:   acl.Entry.PermissionType.String(),
-			Operation:    acl.Entry.Operation.String(),
-			ResourceType: acl.Pattern.ResourceType.String(),
-			ResourceName: acl.Pattern.Name,
-			PatternType:  acl.Pattern.PatternType.String(),
+			Principal:    principal,
+			Permission:   acl.GetPermission(),
+			Operation:    acl.GetOperation(),
+			ResourceType: string(acl.GetResourceType()),
+			ResourceName: acl.GetResourceName(),
+			PatternType:  acl.GetPatternType(),
 		})
 	}
 	list.Filter(listFields)
 	return list.Print()
 }
 
-func getPrefixAndResourceIdFromPrincipal(principal string, numericIdToResourceId map[int32]string) (string, string, error) {
+func principalHasIntegerID(principal string) (bool, error) {
 	if principal == "" {
-		return "", "", nil
+		return false, nil
 	}
 
 	x := strings.Split(principal, ":")
 	if len(x) < 2 {
-		return "", "", errors.Errorf("unrecognized principal format %s", principal)
+		return false, errors.Errorf("unrecognized principal format %s", principal)
 	}
-	prefix := x[0]
 	suffix := x[1]
 
-	// The principal has a resource ID
-	resources := []string{
-		resource.IdentityPool,
-		resource.ServiceAccount,
-		resource.User,
-	}
-	if types.Contains(resources, resource.LookupType(suffix)) {
```

Comment:
for those normal acl's we don't validate the resource type anymore?

Reply:
we might still wanna keep that. (it might just be github not showing the code change properly?

Reply:
This is just a diff from reverting.

The main goal of this part was to determine if we needed to translate it or not (whether or not it had a resource id already). We don't want to do that anymore even for the leftover integer IDs being returned, so we don't need this check.

Reply:
Also this whole diff disappeared after I fixed a merge conflict...